hazelcast/src/main/java/com/hazelcast/nio/Bits.java

```diff
@@ -337,4 +348,34 @@ public static long combineToLong(int x, int y) {
     public static int extractInt(long value, boolean lowerBits) {
         return (int) ((lowerBits) ? value : (value >> 32));
     }
+
+    /**
+     * Fast method of finding the next power of 2 greater than or equal to the supplied value.
+     * <p/>
+     * If the value is &lt;= 0 then 1 will be returned.
+     * <p/>
+     * This method is not suitable for {@link Integer#MIN_VALUE} or numbers greater than 2^30.
+     *
+     * @param value from which to search for next power of 2
+     * @return The next power of 2 or the value itself if it is a power of 2
+     */
+    public static int findNextPositivePowerOfTwo(final int value) {
```

Comment:
Inconveniently, `nextPowerOfTwo` is `long`-based and my version was `int`-based. This will cause trouble at the highest bit (that's why it was  called "find next _positive_ power of two".